modules/acquia_cms_headless/acquia_cms_headless.install

```diff
@@ -21,6 +21,13 @@ function acquia_cms_headless_install() {
   // Adds the necessary permissions to roles.
   add_next_site_permissions();
 
+  // Set the iFrame domain in Media settings with the site base URL.
+  $media_settings = \Drupal::configFactory()->getEditable('media.settings');
+  $iframe_domain = $media_settings->get('iframe_domain');
+  if (empty($iframe_domain)) {
+    $host = \Drupal::request()->getSchemeAndHttpHost();
+    $media_settings->set('iframe_domain', $host)->save();
+  }
```

Comment:
This setting will update the iframe domain on the domain setting. We have updated this setting and have not used any other way to add the domain to media oembed asset because of the following reason - 
If we se the domain using any other method apart from the media settings then the method will have to process the complete page to look for media oembed assets, then identify the asset as something that is to be used by an API and then handle the domain changes according to the site (Drupal or Headless next site). This may not be a performant method.

Whereas, media setting sets the domain while the media is created.